File: CS/PropertyGridMenuOpening/Issues.cs
using DevExpress.Mvvm.DataAnnotations;
using System.Collections.Generic;

namespace PropertyGridMenuOpening {
    public class IssueView {
        public string ProductName { get; set; }
        [NewItemInstanceInitializer(typeof(Bug))]
        [NewItemInstanceInitializer(typeof(Question))]
        [NewItemInstanceInitializer(typeof(BreakingChange))]
        public IssueList Issues { get; internal set; }
    }
    public class IssueList : List<Issue> {
        public override string ToString() {
            return "Issues";
        }
    }
    public abstract class Issue {
        public string Header { get; set; }
        public string Owner { get; set; }

    }
    public enum Severity {
        Minor,
        Moderate,
        Severe
    }
    public class Bug : Issue {
        public Severity Severity { get; set; }
        public string Assignee { get; set; }
    }
    public class Question : Issue {
        public string Assignee { get; set; }
    }
    public enum ChangeType {
        APIChange,
        UIChange,
        BehaviorChange
    }
    public class BreakingChange : Issue {
        public ChangeType ChangeType { get; set; }
    }
}

File: VB/PropertyGridMenuOpening/Issues.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports System.Collections.Generic

Namespace PropertyGridMenuOpening

    Public Class IssueView

        Private _Issues As IssueList

        Public Property ProductName As String

        <NewItemInstanceInitializer(GetType(Bug))>
        <NewItemInstanceInitializer(GetType(Question))>
        <NewItemInstanceInitializer(GetType(BreakingChange))>
        Public Property Issues As IssueList
            Get
                Return _Issues
            End Get

            Friend Set(ByVal value As IssueList)
                _Issues = value
            End Set
        End Property
    End Class

    Public Class IssueList
        Inherits List(Of Issue)

        Public Overrides Function ToString() As String
            Return "Issues"
        End Function
    End Class

    Public MustInherit Class Issue

        Public Property Header As String

        Public Property Owner As String
    End Class

    Public Enum Severity
        Minor
        Moderate
        Severe
    End Enum

    Public Class Bug
        Inherits Issue

        Public Property Severity As Severity

        Public Property Assignee As String
    End Class

    Public Class Question
        Inherits Issue

        Public Property Assignee As String
    End Class

    Public Enum ChangeType
        APIChange
        UIChange
        BehaviorChange
    End Enum

    Public Class BreakingChange
        Inherits Issue

        Public Property ChangeType As ChangeType
    End Class
End Namespace
